src/event-stream/snp-event-stream.ts

```diff
@@ -0,0 +1,75 @@
+import { SERVER_VERSION } from '@hirosystems/api-toolkit';
+import { logger as defaultLogger } from '@hirosystems/api-toolkit';
+import { StacksEventStream, StacksEventStreamType } from '@hirosystems/salt-n-pepper-client';
+import { EventEmitter } from 'node:events';
+import { EventStreamServer } from './event-server';
+import { PgWriteStore } from '../datastore/pg-write-store';
+
+export class SnpEventStreamHandler {
+  db: PgWriteStore;
+  eventServer: EventStreamServer;
+  logger = defaultLogger.child({ name: 'SnpEventStreamHandler' });
+  snpClientStream: StacksEventStream;
+  redisUrl: string;
+  redisStreamPrefix: string | undefined;
+
+  readonly events = new EventEmitter<{
+    processedMessage: [{ msgId: string }];
+  }>();
+
+  constructor(opts: { db: PgWriteStore; eventServer: EventStreamServer; lastMessageId: string }) {
+    this.db = opts.db;
+    this.eventServer = opts.eventServer;
+
+    this.redisUrl = process.env.SNP_REDIS_URL as string;
+    if (!this.redisUrl) {
+      throw new Error('SNP_REDIS_URL environment variable is not set');
+    }
+
+    this.redisStreamPrefix = process.env.SNP_REDIS_STREAM_KEY_PREFIX;
+
+    this.logger.info(`SNP streaming enabled, lastMsgId: ${opts.lastMessageId}`);
+
+    const appName = `stacks-blockchain-api ${SERVER_VERSION.tag} (${SERVER_VERSION.branch}:${SERVER_VERSION.commit})`;
+
+    this.snpClientStream = new StacksEventStream({
+      redisUrl: this.redisUrl,
+      redisStreamPrefix: this.redisStreamPrefix,
+      eventStreamType: StacksEventStreamType.all,
+      lastMessageId: opts.lastMessageId,
+      appName,
+    });
+  }
+
+  async start() {
+    this.logger.info(`Connecting to SNP event stream at ${this.redisUrl} ...`);
+    await this.snpClientStream.connect({ waitForReady: true });
+    this.snpClientStream.start(async (messageId, timestamp, path, body) => {
+      return this.handleMsg(messageId, timestamp, path, body);
+    });
+  }
+
+  async handleMsg(messageId: string, timestamp: string, path: string, body: any) {
+    this.logger.debug(`Received SNP stream event ${path}, msgId: ${messageId}`);
+
+    const response = await this.eventServer.fastifyInstance.inject({
+      method: 'POST',
+      url: path,
+      payload: body,
+    });
```

Comment:
The SNP event streaming implementation is currently layered on top of the existing http event observer code, as you can see here with the `fastify.inject(...)`. There's a significant amount of code in the event observer around logging, IDB, etc which is able to be reused here without a significant refactoring and/or code duplication. 

I _think_ this is fine, but probably not the most optimized in terms of memory usage. I'm assuming this approach results in event bodies being copied in memory at least once.

Reply:
I think this is good tradeoff for now IMO, we could revisit this decision if we see significant memory crunch in our writer